### PR TITLE
Check outside of the ViewPatch.Load body if the first revision is given.

### PR DESCRIPTION
**Pull request by jbialobr**

### CODE REVIEW

File: GitUI/GitUIExtensions.cs [hunk 1]
```diff
@@ -86,20 +86,30 @@ public static Task ViewChanges(this FileViewer diffViewer, IList<GitRevision> re
...
+                if (file.TreeGuid.IsNullOrEmpty())
                 {
-                    // The previous commit does not exist, nothing to compare with
-                    if (file.TreeGuid.IsNullOrEmpty())
-                        diffViewer.ViewGitItemRevision(file.Name, secondRevision);
-                    else
-                        diffViewer.ViewGitItem(file.Name, file.TreeGuid);
+                    if (secondRevision.IsNullOrWhiteSpace())
+                    {
+                        throw new ArgumentException(nameof(secondRevision));
```
Comment (gerhardol) [inline GitUI/GitUIExtensions.cs:96]: I would like to return defaultText in 2.51, suggest investigation #4586 is done first.

Comment (jbialobr) [reply GitUI/GitUIExtensions.cs:96]: I don't understand your comment. Could you please clarify it?

**Comment by gerhardol** [comment]
Thanks
I was not confident to make the Task, async/await changes but were restoring the functionality from 2.50 so it could be safely applied to the 2.51 branch. (In 2.50 also new files were presented as files, not a diff to /dev/null, so not exactly the same).

File: GitUI/GitUIExtensions.cs (hunk 1 above)
Comment (gerhardol) [reply GitUI/GitUIExtensions.cs:96]: The use of null as first or second revision is not consistent. In some situations the revisions are null when comparing. I would like to review that handling, suggest to not merge to 2.51 (or use the initial commit in 2.51).

**Pull request merged**